scripts/test/run.py

```diff
@@ -95,13 +95,49 @@ def nightly_pandas():
     subprocess.run(cmd, check=True)
 
 
-def released_pandas():
-    # query pandas version
+def _get_version_from_pyproject(program: str) -> str:
     text = Path("pyproject.toml").read_text()
     version_line = next(
-        line for line in text.splitlines() if line.startswith("pandas = ")
+        line for line in text.splitlines() if line.startswith(f"{program} = ")
     )
-    version = version_line.split('"')[1]
+    return version_line.split('"')[1]
+
 
+def released_pandas():
+    version = _get_version_from_pyproject("pandas")
     cmd = [sys.executable, "-m", "pip", "install", f"pandas=={version}"]
     subprocess.run(cmd, check=True)
+
+
+def nightly_mypy():
+    cmd = [
+        sys.executable,
+        "-m",
+        "pip",
+        "install",
+        "--upgrade",
+        "git+https://github.com/python/mypy.git",
+    ]
+    subprocess.run(cmd, check=True)
+
+    # ignore unused ignore errors
+    config_file = Path("pyproject.toml")
+    config_file.write_text(
```

Comment:
I don't really like changing pyproject.toml. I'll see whether there is a better way.

I'm also open to not run it on the CI - if there are errors that cannot be resolved by `warn_unused_ignores`, it would be quite annoying to get a failed email for each PR.

Reply:
> I don't really like changing pyproject.toml. I'll see whether there is a better way.

We could put the options used for the nightly tests in a `mypy_nightly.ini` file and then do nightly with the `mypy` option `--config-file mypy_nightly.ini` .  Then this allows us to turn off/on different options for the nightly.

> 
> I'm also open to not run it on the CI - if there are errors that cannot be resolved by `warn_unused_ignores`, it would be quite annoying to get a failed email for each PR.

If we use a special config file, then we just have to modify that config file